### PR TITLE
Adds 'Bloody Spreader' component that bloodies everything it touches

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Adds 'Bloody Spreader' component that bloodies everything it touches!

For example inserting an item into it if it is a storage item. Or entering it if it's a turf, or bumping onto it, or... you get the point, hopefully.

Added this component to the MEAT backpack, meat slabs, bouncy castles, meateor fluff, meateor heart, and the heretic sanguine blade. 

Gave most of these the blood walk component as well, which spreads blood if it's dragged around.

Meat slabs contain a limited amount of both components, eventually they will 'dry out'.
## Why It's Good For The Game

Meat isn't meaty and squelchy enough, this will make it meatier. It also makes the janitor suffer.
## Changelog
:cl:
code: Adds 'Bloody Spreader' component that bloodies everything it touches!
code: For example inserting an item into it if it is a storage item. Or entering it if it's a turf, or bumping onto it, or... you get the point, hopefully.
add: Added this component to the MEAT backpack, meat slabs, bouncy castles, meateor fluff, meateor heart, and the heretic sanguine blade. 
add: Gave most of these the blood walk component as well, which spreads blood if it's dragged around.
add: Meat slabs contain a limited amount of both components, eventually they will 'dry out'.
code: Added a signal for when an item is entered into storage.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
beefmen component

File: code/modules/mob/living/basic/space_fauna/meteor_heart/meteor_heart.dm [hunk 1]
```diff
@@ -60,6 +60,13 @@
 	soundloop.pressure_affected = FALSE
 	soundloop.start()
 
+	. = ..()
+
+	AddComponent(/datum/component/bloody_spreader,\
+		blood_left = INFINITY,\
+		blood_dna = list("meaty DNA" = "MT-"),\
+		diseases = null,\
+	)
```
Comment (carlarctg) [inline code/modules/mob/living/basic/space_fauna/meteor_heart/meteor_heart.dm:69]:
```suggestion

	AddComponent(/datum/component/bloody_spreader,\
		blood_left = INFINITY,\
		blood_dna = list("meaty DNA" = "MT-"),\
		diseases = null,\
	)
```

File: code/datums/components/bloody_spreader.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/datum/component/bloody_spreader
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	// How many bloodening instances are left. Deleted on zero.
+	var/blood_left
+	// We will spread this blood DNA to targets!
+	var/list/blood_dna
+	// For the love of god keep this null
+	var/list/diseases
```
Comment (Ghommie) [inline code/datums/components/bloody_spreader.dm:8]: What's the point of this var if it has to stay null? That's not a nice comment (also it should be `///` for documentation purposes iirc)

File: code/datums/components/bloody_spreader.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	RegisterSignals(parent, signals_to_add, PROC_REF(spread_yucky_blood))
+
+	if(isclothing(parent))
+		parent.AddComponent(/datum/component/bloodysoles)
+
+	src.blood_left = blood_left
+	src.blood_dna = blood_dna
+	src.diseases = diseases
+
+/datum/component/bloody_spreader/proc/spread_yucky_blood(atom/parent, atom/bloody_fool)
```
Comment (Ghommie) [inline code/datums/components/bloody_spreader.dm:33]: Have you checked that all of the signals have an atom as second argument?

**Comment by Loafin34** [comment]
add this onto limbs. viscera cleanup detail experience

File: code/datums/components/bloody_spreader.dm (hunk 3 above)
Comment (carlarctg) [reply code/datums/components/bloody_spreader.dm:33]: It's not a necessary argument, just helpful

Comment (carlarctg) [reply code/datums/components/bloody_spreader.dm:33]: yes

**Comment by zombiegums** [comment]
Minor problem when playing as cook. Bloody Spreader persists even after cooking meat. An example would be grilling bacon: handling the bacon makes you bloody, and it spreads everywhere if you throw the cooked bit of meat, too.

**Pull request merged**